data/blog/enabling-out-of-the-box-performance-insights-in-the-unity-sdk.md

```diff
@@ -0,0 +1,188 @@
+---
+title: 'Enabling Out-of-the-Box Performance Insights in Unity Games with the Sentry SDK'
+date: '2024-11-04'
+tags: ['unity', 'sdk', 'c#']
+draft: false
+summary: 'Learn how we built the autoinstrumentation in the Unity SDK via IL Weaving'
+images: []
+layout: PostLayout
+canonicalUrl:
+authors: []
+---
+
+## Introduction: From Crash Reporting to Performance Insights
+
+Our Unity SDK was super complete from the crash reporting point of view. It had support for line numbers in C# exceptions on IL2CPP (in release mode!), captured native crashes on Windows, macOS, Linux Android and iOS, context set via C# would show up on any type of event, including minidumps, debug symbols are magically uploaded when you build the game with the editor. And more. We were confident we had the best crash reporting solution out there. Now we were looking towards offering some out-of-the-box insights into the game’s performance. Right out of the gate we hit the first question: What would auto-instrumentation for Unity games look like?
+
+## Adapting Sentry's Performance UX for Unity
+
+Sentry had built UX for visualization of span trees and the instrumentation for mobile and web is based around screen rendering. We wanted to take those concepts and apply them to Unity. As a result we limited the instrumentation to the game’s startup procedure and scene loading. Every game starts at some point. And every game, no matter how big or small, loads a scene. We might not be able to give insights into the entire game but we can show every developer right after installing the package, what Sentry could offer.
+
+Our ideal scenario would be something that would work out-of-the-box with no to minimal setup from the user. As a sneak-peek and to show off what we got working without you having to read the whole thing before you get as excited as we are: This is what the Unity SDK‘s auto-instrumentation offers OOTB right now. Without a single line of code. For all Unity games.
+
+![TraceView](/images/autoinstrumentation-in-unity-via-ilweaving/traceview.png)
```

Comment:
path created is: `/images/autoinstrumentation-in-unity-via-ilweaving/traceview.png` which seems right:
![image](https://github.com/user-attachments/assets/13007021-1f30-4425-9305-086e0d3186fe)
![image](https://github.com/user-attachments/assets/dda9d066-82ed-4a35-82aa-cf841da77187)

But on the preview build it doesn't load